HaxeManual/10-std.tex

```diff
@@ -597,6 +597,170 @@ \subsection{Serialization format}
+	// iterator on all attributes.
+}
+\end{lstlisting}
+
+\subsection{Parsing Xml}
+\label{std-Xml-parsing}
+
+Use the \expr{Xml.parse} static method to parse \emph{XML} data and obtain a Haxe value from it:
+
+\begin{lstlisting}
+var xml = Xml.parse('<root>Haxe is great!</root>').firstElement();
+trace(xml.firstChild().nodeValue);
+\end{lstlisting}
+
+\subsection{Encoding Xml}
+\label{std-Xml-encoding}
+
+Use the \expr{xml.toString()} method to encode a Haxe value into a \emph{XML} string:
+\begin{lstlisting}
+var xml = Xml.createElement('root');
+xml.addChild(Xml.createElement('child1'));
+xml.addChild(Xml.createElement('child2'));
+
+trace(xml.toString()); // <root><child1/><child2/></root>
+\end{lstlisting}
+
+\subsection{Fast Xml access}
+\label{std-Xml-Fast}
+
+While the base \type{Xml} API is easy to use, it is sometimes a bit painful to do all the checks of \type{Xml} correctness and can result in verbose code. The \href{http://api.haxe.org/haxe/xml/Fast.html}{haxe.xml.Fast} API is here to help by providing a fast dot-syntax access to the most common \type{Xml} methods.
+
+Here's an example of fast XML usage:
+\begin{lstlisting}
+// parse some XML data
+var xml = Xml.parse("
+  <user name='John'>
+    <phone>
+      <number>0000</number>
+      <number>111</number>
+    </phone>
+  </user>
+");
+// wrap the Xml for fast access
+var fast = new haxe.xml.Fast(xml.firstElement());
+
+// access attributes
+trace(fast.att.name); // attribute "name"
+if (fast.has.age) trace( fast.att.age ); // optional attribute
+
+// access the "phone" child, which is wrapped with haxe.xml.Fast too
+var phone = fast.node.phone;
+
+// iterate over numbers
+for (p in phone.nodes.number) {
+  trace(p.innerData);
+}
+\end{lstlisting}
+This code works the same on all platforms.
+
+\paragraph{Accessors}
+There are different accessors that can be used with the Fast API:
+\begin{description}
+	\item[\expr{.name}] returns the name of the current element (same as \expr{Xml.nodeName}).
+	\item[\expr{.x}] returns the current corresponding \type{Xml} node.
+	\item[\expr{.att.<name>}] access to a given attribute. An exception is thrown if the attribute doesn't exists.
+	\item[\expr{.has.<name>}] check the existence of an attribute.
+	\item[\expr{.elements}] the list of all sub-elements (which are the nodes with type \type{Xml.Element}).
+	\item[\expr{.node.<name>}] access to the first sub element with the given name. An exception is thrown if the element doesn't exists.
+	\item[\expr{.nodes.<name>}] returns a List of elements with the given name.
+	\item[\expr{.hasNode.<name>}] check the existence of a sub node with the given name.
+	\item[\expr{.innerData}] returns the inner \ic{PCDATA} or \ic{CDATA} of the node. An exception is thrown if there is no data or if there not only data but also other nodes.
+	\item[\expr{.innerHTML}] returns the XML string built with all the sub nodes, excluding the current one.
+\end{description}
+
+\paragraph{Self-closed nodes}
```

Comment:
This entire section confuses me, I'm not quite sure what it is about and why this would be a special case.